auth-server/src/services/auth.ts

```diff
@@ -160,6 +244,87 @@ export default class AuthService {
 			throw new ForbiddenError(messages.ADDRESS_LOGIN_NOT_FOUND);
 		}
 
+		await redisDel(getAddressLoginKey(address));
+
+		return {
+			refreshToken: await this.getRefreshToken(user),
+			token: await this.getSignedToken(user),
+			user: {
+				email: user.email,
+				email_verified: user.email_verified || false,
+				id: user.id,
+				name: user.name,
+				username: user.username,
+				web3signup: user.web3signup || false
+			}
+		};
+	}
+
+	public async AddressSignupStart (address: string): Promise<string> {
+		const addressObj = await Address
+			.query()
+			.where('address', address)
+			.first();
+
+		if (addressObj) {
+			throw new ForbiddenError(messages.ADDRESS_SIGNUP_ALREADY_EXISTS);
+		}
+
+		const signMessage = uuid();
+
+		await redisSetex(getAddressSignupKey(address), ADDRESS_LOGIN_TTL, signMessage);
+
+		return signMessage;
+	}
+
+	public async AddressSignupConfirm (network: string, address: string, signature: string, email: string, username: string, name: string): Promise<AuthObjectType> {
+		const signMessage = await redisGet(getAddressSignupKey(address));
+
+		if (!signMessage) {
+			throw new ForbiddenError(messages.ADDRESS_SIGNUP_SIGN_MESSAGE_EXPIRED);
+		}
+
+		const isValidSr = verifySignature(signMessage, address, signature);
+
+		if (!isValidSr) {
+			throw new ForbiddenError(messages.ADDRESS_SIGNUP_INVALID_SIGNATURE);
+		}
+
+		const addressObj = await Address
+			.query()
+			.where('address', address)
+			.first();
+
+		if (addressObj) {
+			throw new ForbiddenError(messages.ADDRESS_SIGNUP_ALREADY_EXISTS);
+		}
+
+		let existing = await User
```

Comment:
This will be removed, and it's actually good because there's nothing that enforces the unicity of the on-chain name. this means that the signup might have failed without any way for our users to know that it came from the identity.

Reply:
removed